thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -365,19 +365,18 @@ class ThinReplicaImpl {
             update_aggregator_counter = 0;
           }
         }
-        if (!is_event_group_transition) {
-          config_->subscriber_list.removeBuffer(live_updates);
-          live_updates->removeAllUpdates();
-          metrics_.subscriber_list_size.Get().Set(config_->subscriber_list.Size());
-          metrics_.updateAggregator();
-        }
       } catch (std::exception& error) {
         LOG_INFO(logger_, "Subscription stream closed: " << error.what());
       }
-      if (context->IsCancelled()) {
-        return grpc::Status::CANCELLED;
-      }
-      if (not is_event_group_transition) {
+      // Clean up if we need to return
+      if (not is_event_group_transition || context->IsCancelled()) {
+        config_->subscriber_list.removeBuffer(live_updates);
+        live_updates->removeAllUpdates();
+        metrics_.subscriber_list_size.Get().Set(config_->subscriber_list.Size());
+        metrics_.updateAggregator();
+        if (context->IsCancelled()) {
+          return grpc::Status::CANCELLED;
```

Comment:
Nit: I think a log message `LOG_WARN(logger_, "Subscription cancelled");` like we have in `TryPopEventGroup()` can be helpful for debugging.

Reply:
This gets triggered more often than not if there are no updates. Having a warning message would actually spam the logs. We can do a debug message but even then it would be pure spam :/